client/extensions/woocommerce/index.js

```diff
@@ -203,21 +202,22 @@ function addStorePage( storePage, storeNavigation ) {
 				appProps.documentTitle = storePage.documentTitle;
 			}
 
-			let analyticsPath = storePage.path;
-			const { filter } = context.params;
+			appProps.analyticsPath = storePage.path.replace( '?', '' );
+			const { filter, productId } = context.params;
 			if ( filter ) {
-				analyticsPath = analyticsPath.replace( ':filter', filter );
+				appProps.analyticsPath = appProps.analyticsPath.replace( ':filter', filter );
+			}
+			if ( productId ) {
+				appProps.analyticsPath = appProps.analyticsPath.replace( ':productId', ':product-id' );
```

Comment:
Are these being set somewhere else? Would it be better to change where they are originally set, rather than doing this here?

Reply:
Yup, they're defined here: https://github.com/Automattic/wp-calypso/blob/cc8bc840b799dadb6e2eadb39cf8bde37582c91d/client/extensions/woocommerce/index.js#L47

Basically it's a quite interesting approach of defining an array of paths, and then feed them to `page`.
It turns out very handy for our page views work most of the times.
Though in some cases I've preferred doing some string replacement than delving deeper.

For example, `:productId` is passed as a `productId` prop to a main component.
Changing it into `:product-id` in the actual path would have also meant changing it in the component, which is not something I'm keen to do. 🙂